internal/provider/models/processors/trace_sampling.go

```diff
@@ -0,0 +1,250 @@
+			Validators: []validator.Int64{
+				int64validator.AtLeast(2),
+				int64validator.AtMost(100000),
+			},
+		},
+		"parent_span_id_field": schema.StringAttribute{
+			Optional:    true,
+			Computed:    true,
+			Description: "Tail Sampling: The field name to pull the parent span id. Absence of this field indicates the event is a root span.",
+		},
+		"conditionals": schema.ListNestedAttribute{
+			Optional:    true,
+			Description: "Tail Sampling: A list of complete conditionals, including expressions and a sample rate to maintain.",
+			NestedObject: schema.NestedAttributeObject{
+				Attributes: (map[string]schema.Attribute{
+					"conditional": schema.SingleNestedAttribute{
+						Computed:    true,
+						Optional:    true,
+						Description: "Tail Sampling: A conditional expression for sampling.",
+						Attributes:  ExtendSchemaAttributes(modelutils.ParentConditionalAttribute(modelutils.Non_Change_Operator_Labels).Attributes, map[string]schema.Attribute{}),
+					},
+					"rate": schema.Int64Attribute{
+						Required:    true,
+						Description: "The rate at which to sample traces expressed as 1/N. A value of 1 means 100% sampling, 2 means 50%, 3 means 33%, etc.",
+						Validators: []validator.Int64{
+							int64validator.AtLeast(1),
+							int64validator.AtMost(100000),
+						},
+					},
+					"_output_name": schema.StringAttribute{
+						Computed:    true,
+						Optional:    true,
+						Description: "The generated name of the rule.",
+					},
+				}),
+			},
+		},
+	}),
+}
+
+func TraceSamplingProcessorFromModel(plan *TraceSamplingProcessorModel, previousState *TraceSamplingProcessorModel) (*client.Processor, diag.Diagnostics) {
+	dd := diag.Diagnostics{}
+
+	// a lot of config props are marked optional, so we need to check if they are set
+	// because they may not actually be optional or applicable given the type
+	if !ValidateRequiredConfigsPerSampleType(plan, plan.SampleType.ValueString(), &dd) {
+		return nil, dd
+	}
+
+	component := client.Processor{
+		BaseNode: client.BaseNode{
+			Type:        TRACE_SAMPLING_PROCESSOR_NODE_NAME,
+			Title:       plan.Title.ValueString(),
+			Description: plan.Description.ValueString(),
+		},
+	}
+	if previousState != nil {
+		component.Id = previousState.Id.ValueString()
+		component.GenerationId = previousState.GenerationId.ValueInt64()
+	}
+	component.Inputs = modelutils.StringListValueToStringSlice(plan.Inputs)
+	component.UserConfig = map[string]any{
+		"sample_type": plan.SampleType.ValueString(),
+	}
+	if !plan.TraceIdField.IsNull() && plan.TraceIdField.ValueString() != "" {
+		component.UserConfig["trace_id_field"] = plan.TraceIdField.ValueString()
+	}
+
+	if plan.SampleType.ValueString() == "head" {
+		if !plan.Rate.IsNull() && plan.Rate.ValueInt64() != 0 {
+			component.UserConfig["rate"] = plan.Rate.ValueInt64()
+		}
+	} else {
+		if !plan.ParentSpanIdField.IsNull() && plan.ParentSpanIdField.ValueString() != "" {
```

Comment:
We can also replace the `!= ""` check with a min length validation on the parent span ID field